src/charm.py

```diff
@@ -48,17 +48,31 @@ def _store_tls_resources(
         """Write certificates and keys on disk."""
         certs_dir = self.opensearch.paths.certs
 
+        if not secrets.get("key"):
```

Comment:
We should not test on this, we should rather let it crash if not exists. Any specific reason why we're checking on this?

Reply:
Previously we allowed for `None` values. I.e. having a dictionary key undefined wasn't equal to the same key having a `None` value. Thus I tried to securize such calls making sure that behavior of the charm would be the same on Juju 2 and Juju 3.

I can change the code to raise an exception if `key` is not defined, in case preferred.

Reply:
Do nothing - key cannot be None (gets generated before certificates).